src/account/UpgradeableModularAccount.sol

```diff
@@ -305,7 +300,7 @@ contract UpgradeableModularAccount is
         onlyProxy
         wrapNativeFunction
     {
-        _upgradeToAndCallUUPS(newImplementation, data, true);
+        upgradeToAndCall(newImplementation, data);
```

Comment:
I just realized - does this recurse with the internal function definition? Our goal is to invoke the version of the function as defined in `UUPSUpgradeable`, right? So would we need to call `super.upgradeToAndCall`?

Reply:
yep, need either `super.upgradeToAndCall` or `UUPSUpgradeable.upgradeToAndCall`. Added a test at the same time!